internal/pkg/kubernetes/client.go

```diff
@@ -0,0 +1,360 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"octavius/internal/executor/config"
+	"octavius/internal/pkg/constant"
+	"os"
+	"path/filepath"
+	"time"
+
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	typeMeta     meta.TypeMeta
+	ctx          context.Context
+	cancel       context.CancelFunc
```

Comment:
why put cancel here? I don't recommend cancel used as singleton, you should pass the cancel over the function.

Reply:
I saw many usages of cancel referring to this var, it will encounter race condition because many functions refer to one ctx and cancel var